test/UnitTests/MSTest.CoreAdapter.Unit.Tests/Execution/TestMethodInfoTests.cs

```diff
@@ -131,6 +131,70 @@ public void TestMethodInfoInvokeShouldWaitForAsyncTestMethodsToComplete()
         }
 
         [TestMethodV1]
+        public void TestMethodInfoInvokeAsyncShouldHandleThrowAssertInconclusive()
+        {
+            DummyTestClass.DummyAsyncTestMethodBody = () => Task.Run(() => { throw new UTF.AssertInconclusiveException(); });
+            var asyncMethodInfo = typeof(DummyTestClass).GetMethod("DummyAsyncTestMethod");
+
+            var method = new TestMethodInfo(
+                asyncMethodInfo,
+                this.testClassInfo,
+                this.testMethodOptions);
+
+            var result = method.Invoke(null);
+
+            Assert.AreEqual(UTF.UnitTestOutcome.Inconclusive, result.Outcome);
+        }
+
+        [TestMethodV1]
+        public void TestMethodInfoInvokeAsyncShouldHandleAssertInconclusive()
+        {
+            DummyTestClass.DummyAsyncTestMethodBody = () => Task.Run(() => { UTF.Assert.Inconclusive(); });
+            var asyncMethodInfo = typeof(DummyTestClass).GetMethod("DummyAsyncTestMethod");
+
+            var method = new TestMethodInfo(
+                asyncMethodInfo,
+                this.testClassInfo,
+                this.testMethodOptions);
+
+            var result = method.Invoke(null);
+
+            Assert.AreEqual(UTF.UnitTestOutcome.Inconclusive, result.Outcome);
+        }
+
+        [TestMethodV1]
+        public void TestMethodInfoInvokeShouldHandleThrowAssertInconclusive()
+        {
+            DummyTestClass.TestMethodBody = (d) => { throw new UTF.AssertInconclusiveException(); };
+            var asyncMethodInfo = typeof(DummyTestClass).GetMethod("DummyAsyncTestMethod");
```

Comment:
1. We would want to do GetMethod on "DummyTestMethod" here..
2. Should we just name the varuable as "methodInfo" rather than "asyncMethodInfo" ?

Reply:
Yes and almost. methodInfo was a class member, so I've made it "dummyMethodInfo" instead.